### PR TITLE
Revert "Automatically track subloggers in allLoggers (#22038)"

### PR DESCRIPTION
**Pull request by hghaf099**
This reverts commit 4c8cc87794ed2d989f515cd30c1c1b953d092ef3.

### CODE REVIEW

File: vault/logical_system_test.go [hunk 1]
```diff
@@ -5665,13 +5666,6 @@ func TestSystemBackend_LoggersByName(t *testing.T) {
 			true,
 			false,
 		},
-		{
```
Comment (hghaf099) [inline vault/logical_system_test.go:5668]: Checking why this was removed

Comment (raskchanky) [reply vault/logical_system_test.go:5668]: The original PR added it. I think it's fine.

**Review by raskchanky — APPROVED** [review]

**Review by mpalmi — APPROVED** [review]

**Comment by hghaf099** [comment]
Backport PRs: 
1.13: https://github.com/hashicorp/vault/pull/24004
1.14: https://github.com/hashicorp/vault/pull/24005
1.15: https://github.com/hashicorp/vault/pull/24003

A bit I missed from reverting the PR in main:
https://github.com/hashicorp/vault/pull/24006

**Pull request merged**